commonv5/ts/helpers/request.ts

```diff
@@ -1,68 +1,41 @@
 import * as tl from "azure-pipelines-task-lib/task";
-import * as request from "request";
+import fetch from "node-fetch";
 import * as semver from "semver";
 import Endpoint from "../sonarqube/Endpoint";
 
 interface RequestData {
   [x: string]: any;
 }
 
-function get(endpoint: Endpoint, path: string, isJson: boolean, query?: RequestData): Promise<any> {
+export async function get<T>(
+  endpoint: Endpoint,
+  path: string,
+  isJson: boolean,
+  query?: RequestData,
+): Promise<T | string> {
   tl.debug(`[SQ] API GET: '${path}' with query "${JSON.stringify(query)}"`);
-  return new Promise((resolve, reject) => {
-    const options: request.CoreOptions = {
-      auth: endpoint.auth,
-      timeout: 60000,
-    };
-    if (query) {
-      options.qs = query;
-      options.useQuerystring = true;
-    }
-    request.get(
-      {
-        method: "GET",
-        baseUrl: endpoint.url,
-        uri: path,
-        json: isJson,
-        ...options,
-      },
-      (error, response, body) => {
-        if (error) {
-          return logAndReject(
-            reject,
-            `[SQ] API GET '${path}' failed, error was: ${JSON.stringify(error)}`,
-          );
-        }
-        tl.debug(
-          `Response: ${response.statusCode} Body: "${
-            isString(body) ? body : JSON.stringify(body)
-          }"`,
-        );
-        if (response.statusCode < 200 || response.statusCode >= 300) {
-          return logAndReject(
-            reject,
-            `[SQ] API GET '${path}' failed, status code was: ${response.statusCode}`,
-          );
-        }
-        return resolve(body || (isJson ? {} : ""));
-      },
-    );
-  });
-}
-
-function isString(x) {
-  return Object.prototype.toString.call(x) === "[object String]";
-}
 
-export function getJSON(endpoint: Endpoint, path: string, query?: RequestData): Promise<any> {
-  return get(endpoint, path, true, query);
-}
-
-export function getServerVersion(endpoint: Endpoint): Promise<semver.SemVer> {
-  return get(endpoint, "/api/server/version", false).then(semver.coerce) as Promise<semver.SemVer>;
+  try {
+    const hasQuery = query && Object.keys(query).length > 0;
+    const response = await fetch(
+      endpoint.url + path + (hasQuery ? "?" + new URLSearchParams(query).toString() : ""),
+      endpoint.toFetch(),
+    );
```

Comment:
We are losing the ability to use a timeout. We could in theory re-add support with signals and abort controller, but not worth it IMHO

Reply:
> Does node fetch have some default timeout?

Actually after double-checking, there indeed is a way to use a timeout - Thanks for spotting, I'll use the `timeout` option